launcher/container_runner_test.go

```diff
@@ -131,6 +169,66 @@ func TestRefreshToken(t *testing.T) {
 	}
 }
 
+// TestRefreshTokenWithSignedContainerCacheEnabled checks `refreshToken` updates the default token when signatures get updated.
+func TestRefreshTokenWithSignedContainerCacheEnabled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	oldCache := []string{"old sigs cache"}
+	fakeAgent := &fakeAttestationAgent{
+		sigsFetcherFunc: func(context.Context) []string {
+			return oldCache
+		},
+		launchSpec: spec.LaunchSpec{Experiments: experiments.Experiments{EnableSignedContainerCache: true}},
+	}
+	fakeAgent.attestFunc = func(context.Context, agent.AttestAgentOpts) ([]byte, error) {
+		return createJWTWithSignatures(t, fakeAgent.sigsCache), nil
+	}
+
+	runner := ContainerRunner{
+		attestAgent: fakeAgent,
+		logger:      log.Default(),
+	}
+
+	if err := os.MkdirAll(launcherfile.HostTmpPath, 0744); err != nil {
```

Comment:
should use https://pkg.go.dev/testing#B.TempDir

Reply:
Since `refreshToken` will always write OIDC token to a fixed location, I think we'll have to create a fixed directory here instead of a random temp test directory.